### ISSUE TITLE
Jacobian computation fails on nn.Module example

### ISSUE THREAD

**Issue by gkaissis** [pytorch/functorch#292]
The following code fails:

```
import torch
from functorch import make_functional, vmap, grad, jacrev
from torch import nn

model = nn.Linear(3, 3).cuda()
data = torch.randn(2, 3).cuda()
targets = torch.randn(2, 3).cuda()

func_model, params = make_functional(model)


def compute_loss(params, data, targets):
    preds = func_model(params, data)
    return torch.mean((preds - targets) ** 2)

jacrev(compute_loss, argnums=(0, 1, 2))(params, data, targets)
```

with the error `'tuple' object has no attribute 'view'` which comes from `eager_transforms` line 364. In reality, I don't even want argnums to be (0,1,2), but if left at its default setting, I'm getting the error `new_args should be of size 1, was of size 2`. 

The error message is not very descriptive. Am I doing something wrong or is this a bug? Also, I've tried vmapping over the batch dimension (`vmap(jacrev(compute_loss), (None, 0, 0))(params, data, targets)`) but this throws the same error.

Congratulations on the amazing work by the way. This is the right way forward for PyTorch!

**Comment by zou3519** [comment]
Hi @gkaissis,

Thank you for the detailed message and the minimal script to reproduce the error! You are not doing something wrong, this is a bug on our side. I know what is wrong, will try to put up a fix for this soon (hopefully tomorrow).

**Comment by zou3519** [comment]
After merging https://github.com/pytorch/functorch/pull/297, the following works for me now:
```
import torch
from functorch import make_functional, vmap, grad, jacrev
from torch import nn

model = nn.Linear(3, 3).cuda()
data = torch.randn(2, 3).cuda()
targets = torch.randn(2, 3).cuda()

func_model, params = make_functional(model)


def compute_loss(params, data, targets):
    preds = func_model(params, data)
    return torch.mean((preds - targets) ** 2)

jacrev(compute_loss, argnums=(0, 1, 2))(params, data, targets)
```
If you reinstall functorch using the main branch, the change will be there. Could you please let us know if that resolves your problem?

**Comment by gkaissis** [comment]
This seems to have fixed the problem and I'm now merrily calculating Jacobians with millions of parameters 🤩 

Fantastic work! Thank you.

**Comment by zou3519** [comment]
Awesome! Out of curiosity, @gkaissis, could you tell us a bit more about your use case? If you have any other feedback around functorch (or bug reports) please do let us know

**Comment by gkaissis** [comment]
Our use-cases are differentially private learning, loss curvature calculations and second-order methods. I'll certainly be following the project with great interest. I think this is one of those projects which have the potential to really advance science because we'll have computational tools to actually get our ideas tested, which is why I am so excited for it. On the matter of feedback, I'd propose making the documentation for the vmap/grad and jacobian computations a bit more intuitive/prominent and easy to find. I found myself spending a lot of time trying to figure out the dimensions over which I want to map and getting incompatible dimension errors. Just a minor thing, but maybe important from a UX standpoint.
Cheers!
